sklearn/neighbors/_nearest_centroid.py

```diff
@@ -161,6 +161,9 @@ def fit(self, X, y):
             # Calculate deviation using the standard deviation of centroids.
             variance = (X - self.centroids_[y_ind]) ** 2
             variance = variance.sum(axis=0)
+            if np.sum(variance) == 0:
+                raise ValueError("All features have zero variance. "
+                                 "Division by zero.")
```

Comment:
We would been to be careful here with using variance directly:

```py
import numpy as np

X = np.empty((10, 2))
X[:, 0] = -0.13725701
X[:, 1] = -0.9853293

X_means = X.mean(axis=0)
var = (X - means)**2
var = var.sum(axis=0)

var
# array([7.70371978e-33, 1.23259516e-31])
```

We can use `np.ptp` instead to look for constant features:

```py
np.ptp(X, axis=0)
# array([0., 0.])
```

Reply:
No problem! Good catch, changed logic to `np.ptp(X).sum() == 0` and changed test case to match your example.